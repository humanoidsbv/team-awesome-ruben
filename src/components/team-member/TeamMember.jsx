import React, { useState } from 'react';
import PropTypes from 'prop-types';

import IconArrowExpand from '../../assets/icons/icon-arrow-expand.svg';
import UserProfileImage from '../../../static/images/picture-tnt.jpg';
import styles from './TeamMember.module.css';

const TeamMember = ({
  teamMember: { currentClient },
  teamMember: { employeeNumber },
  teamMember: { firstName },
  teamMember: { lastName },
  teamMember: { role },
  teamMember: { startingDate }
}) => {
  const [isExpanded, setIsExpanded] = useState(false);

  const handleClick = () => setIsExpanded(!isExpanded);

  return (
    <button
      className={`${styles.container} ${
        isExpanded ? styles.containerActive : ''
      }`}
      onClick={handleClick}
      type="button"
    >
      <img
        alt="Profile avatar"
        className={styles.profileImage}
        src={UserProfileImage}
      />
      <div className={styles.nameContainer}>
        <span className={styles.name}>{`${firstName} ${lastName}`}</span>
        <span className={styles.role}>{role}</span>
      </div>
      <IconArrowExpand
        className={`${styles.arrowExpand}
      ${isExpanded ? styles.arrowExpandActive : ''}`}
      />
      <span
        className={`${styles.descriptionBlock}
      ${isExpanded ? styles.descriptionBlockActive : ''}`}
      >
        {`Detailed information about ${firstName}`}
      </span>
      <div
        className={`
          ${styles.detailContainer} ${styles.employeeNumber}
          ${isExpanded ? styles.detailContainerActive : ''}`}
      >
        <span className={styles.detail}>{employeeNumber}</span>
        <span className={styles.detailDescription}>Employee number</span>
      </div>
      <div
        className={`
          ${styles.detailContainer} ${styles.currentClient}
          ${isExpanded ? styles.detailContainerActive : ''}`}
      >
        <span className={styles.detail}>{currentClient}</span>
        <span className={styles.detailDescription}>Current client</span>
      </div>
      <div
        className={`
          ${styles.detailContainer} ${styles.startingDate}
          ${isExpanded ? styles.detailContainerActive : ''}`}
      >
        <span className={styles.detail}>{startingDate}</span>
        <span className={styles.detailDescription}>Starting date</span>
      </div>
    </button>
  );
};

TeamMember.propTypes = {
  teamMember: PropTypes.shape({
    avatar: PropTypes.string,
    currentClient: PropTypes.string,
    employeeNumber: PropTypes.string,
    firstName: PropTypes.string,
    lastName: PropTypes.string,
    role: PropTypes.string,
    startingDate: PropTypes.string
  })
};

TeamMember.defaultProps = {
  teamMember: []
};

export default TeamMember;
