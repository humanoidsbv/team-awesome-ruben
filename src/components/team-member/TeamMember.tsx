import React, { useState } from 'react';

import IconArrowExpand from '../../assets/icons/icon-arrow-expand.svg';
import UserProfileImage from '../../../static/images/picture-tnt.jpg';
import styles from './TeamMember.module.css';

import { TeamMemberInterface } from '../../redux/team-members/types';

interface TeamMemberProps {
  teamMember: TeamMemberInterface;
}

const TeamMember = ({
  teamMember: {
    currentClient,
    employeeNumber,
    firstName,
    lastName,
    role,
    startingDate
  }
}: TeamMemberProps): React.ReactElement => {
  const [isExpanded, setIsExpanded] = useState(false);

  const handleClick = (): void => setIsExpanded(!isExpanded);

  return (
    <button
      className={`${styles.container} ${
        isExpanded ? styles.containerActive : ''
      }`}
      onClick={handleClick}
      type="button"
    >
      <img
        alt="Profile avatar"
        className={styles.profileImage}
        src={UserProfileImage}
      />
      <div className={styles.nameContainer}>
        <span className={styles.detail}>{`${firstName} ${lastName}`}</span>
        <span className={styles.detailDescription}>{role}</span>
      </div>
      <IconArrowExpand
        className={`${styles.arrowExpand}
      ${isExpanded ? styles.arrowExpandActive : ''}`}
      />
      <span
        className={`${styles.descriptionBlock}
      ${isExpanded ? styles.descriptionBlockActive : ''}`}
      >
        {`Detailed information about ${firstName}`}
      </span>
      <div
        className={`
          ${styles.detailContainer} ${styles.employeeNumber}
          ${isExpanded ? styles.detailContainerActive : ''}`}
      >
        <span className={styles.detail}>{employeeNumber}</span>
        <span className={styles.detailDescription}>Employee number</span>
      </div>
      <div
        className={`
          ${styles.detailContainer} ${styles.currentClient}
          ${isExpanded ? styles.detailContainerActive : ''}`}
      >
        <span className={styles.detail}>{currentClient}</span>
        <span className={styles.detailDescription}>Current client</span>
      </div>
      <div
        className={`
          ${styles.detailContainer} ${styles.startingDate}
          ${isExpanded ? styles.detailContainerActive : ''}`}
      >
        <span className={styles.detail}>{startingDate}</span>
        <span className={styles.detailDescription}>Starting date</span>
      </div>
    </button>
  );
};

export default TeamMember;
