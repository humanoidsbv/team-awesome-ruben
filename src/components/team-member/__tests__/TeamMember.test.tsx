import * as React from 'react';
import { shallow } from 'enzyme';

import TeamMember from '../TeamMember';

const teamMemberMockeddata = {
  currentClient: '',
  employeeNumber: '',
  firstName: '',
  id: 0,
  lastName: '',
  role: '',
  startingDate: ''
};

it('matches component output with expected output', () => {
  const teamMember = shallow(<TeamMember teamMember={teamMemberMockeddata} />);
  expect(teamMember).toMatchSnapshot();
});
