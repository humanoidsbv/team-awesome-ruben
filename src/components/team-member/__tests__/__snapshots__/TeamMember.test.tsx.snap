// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches component output with expected output 1`] = `
<button
  className="container "
  onClick={[Function]}
  type="button"
>
  <img
    alt="Profile avatar"
    className="profileImage"
    src="test-file-stub"
  />
  <div
    className="nameContainer"
  >
    <span
      className="detail"
    >
       
    </span>
    <span
      className="detailDescription"
    />
  </div>
  <test-file-stub
    className="arrowExpand
      "
  />
  <span
    className="descriptionBlock
      "
  >
    Detailed information about 
  </span>
  <div
    className="
          detailContainer employeeNumber
          "
  >
    <span
      className="detail"
    />
    <span
      className="detailDescription"
    >
      Employee number
    </span>
  </div>
  <div
    className="
          detailContainer currentClient
          "
  >
    <span
      className="detail"
    />
    <span
      className="detailDescription"
    >
      Current client
    </span>
  </div>
  <div
    className="
          detailContainer startingDate
          "
  >
    <span
      className="detail"
    />
    <span
      className="detailDescription"
    >
      Starting date
    </span>
  </div>
</button>
`;
